operations/admin_from_mdb.go

```diff
@@ -0,0 +1,282 @@
+					if _, err := doc.ReadFrom(buf); err != nil {
+						if err == io.EOF {
+							break
+						}
+						return errors.Wrap(err, "problem reading document from buffer")
+					}
+					docs = append(docs, doc)
+				}
+				coll.InsertMany(ctx, docs)
+				grip.Infof("inserted %d docs into %s", len(docs), header.Name)
+			}
+
+			return nil
+		},
+	}
+}
+
+func fromMdbForLocal() cli.Command {
+	const (
+		dbFlagName      = "db"
+		outputFlagName  = "output"
+		projectFlagName = "project"
+		urlFlagName     = "url"
+	)
+	collections := []string{"versions", "builds", "tasks", "distro", "project_ref"}
+	return cli.Command{
+		Name:  "dump",
+		Usage: "dump a portion of a MongoDB database for a local Evergreen",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  projectFlagName,
+				Usage: "get data for only this project, to limit data dump",
+			},
+			cli.StringFlag{
+				Name:  outputFlagName,
+				Usage: "write data to this file",
+			},
+			cli.StringFlag{
+				Name:  urlFlagName,
+				Usage: "specify the mongodb url",
+				Value: "mongodb://127.0.0.1:27017",
+			},
+			cli.StringFlag{
+				Name:  dbFlagName,
+				Usage: "read data from this database",
+				Value: "mci",
+			},
+		},
+		Before: mergeBeforeFuncs(
+			func(c *cli.Context) error {
+				if c.String(outputFlagName) == "" {
+					return errors.New("must specify an output file")
+				}
+				if c.String(projectFlagName) == "" {
+					return errors.New("must specify a project to dump data for")
+				}
+				return nil
+			},
+		),
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			outfn := c.String(outputFlagName)
+			project := c.String(projectFlagName)
+			dbName := c.String(dbFlagName)
+			url := c.String(urlFlagName)
+
+			var filters map[string]bson.M
+			if project != "" {
+				filters = map[string]bson.M{
+					"versions": bson.M{"identifier": project},
+					"builds":   bson.M{"branch": project},
+					"tasks":    bson.M{"branch": project},
+				}
+			}
+
+			client, err := mongo.NewClient(options.Client().ApplyURI(url))
+			if err != nil {
+				return errors.Wrap(err, "problem creating mongodb client")
+			}
+
+			connCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+			if err = client.Connect(connCtx); err != nil {
+				return errors.Wrap(err, "problem connecting to mongodb")
+			}
+
+			for _, collection := range collections {
+				coll := client.Database(dbName).Collection(collection)
+				size, err := coll.CountDocuments(ctx, struct{}{})
+				if err != nil {
+					return errors.Wrap(err, "problem finding number of source documents")
+				}
+				if size == 0 {
+					return errors.Errorf("cannot write data from collection ('%s') without documents", collection)
+				}
+			}
+
+			if _, err = os.Stat(outfn); !os.IsNotExist(err) {
+				return errors.Errorf("cannot export to %s, file already exists", outfn)
+			}
+
+			var f io.Writer
+			if outfn != "" {
+				f, err = os.Create(outfn)
```

Comment:
what closes this file? it could be corrupt at the end.

Reply:
My compiler says that `os.Create` returns a `Writer` not a `WriteCloser`. I am surprised by this too.

Reply:
Fixed. I was defining the variable too narrowly.